scripts/generate_driver_wrappers.py

```diff
@@ -22,52 +22,117 @@
 
 import sys
 import os
+import json
+from typing import Tuple
 import argparse
+import jsonschema
+from jsonschema import validate
 import jinja2
 from mbedtls_dev import build_tree
 
-def render(template_path: str) -> str:
+def render(template_path: str, driver_jsoncontext: list) -> str:
     """
-    Render template from the input file.
+    Render template from the input file and driver JSON.
     """
     environment = jinja2.Environment(
         loader=jinja2.FileSystemLoader(os.path.dirname(template_path)),
         keep_trailing_newline=True)
     template = environment.get_template(os.path.basename(template_path))
 
-    return template.render()
+    return template.render(drivers=driver_jsoncontext)
 
-def generate_driver_wrapper_file(mbedtls_root: str, output_dir: str) -> None:
+
+def generate_driver_wrapper_file(template_dir: str, \
+                                 output_dir: str, driver_jsoncontext: list) -> None:
     """
     Generate the file psa_crypto_driver_wrapper.c.
     """
     driver_wrapper_template_filename = \
-        os.path.join(mbedtls_root, \
-        "scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja")
+        os.path.join(template_dir, "psa_crypto_driver_wrappers.c.jinja")
 
-    result = render(driver_wrapper_template_filename)
+    result = render(driver_wrapper_template_filename, driver_jsoncontext)
 
     with open(os.path.join(output_dir, "psa_crypto_driver_wrappers.c"), 'w') as out_file:
         out_file.write(result)
 
+
+def validate_json(driverjson_data: list, driverschema: list) -> bool:
+    """
+    Validate the Driver JSON against schema
+    """
+    try:
+        validate(instance=driverjson_data, schema=driverschema)
+    except jsonschema.exceptions.ValidationError as err:
+        print(err)
+        print("The driver JSON data is InValid")
+        return False
+
+    return True
+
+def merge_driverjsonfiles(mbedtls_root: str, json_directory: str, \
+                          jsondriver_list: str) -> Tuple[bool, list]:
+    """
+    Merge driver JSON files into a single ordered JSON after validation.
+    """
+    result = list()
+    driverlist = list()
+    with open(os.path.join(mbedtls_root, \
+        'scripts/data_files/driver_jsons/driver_transparent_schema.json'), 'r') as file:
+        transparent_driver_schema = json.load(file)
+    with open(os.path.join(mbedtls_root, \
+        'scripts/data_files/driver_jsons/driver_opaque_schema.json'), 'r') as file:
+        opaque_driver_schema = json.load(file)
+
+    with open(os.path.join(json_directory, jsondriver_list), 'r') as driverlistfile:
+        driverlist = json.load(driverlistfile)
```

Comment:
I don't see a major problem here: if `json.load` fails, it'll abort the program with a mostly informative error message, or the calling function can catch the exception if it wants to. The exceptions from JSON parsing do miss one important piece of information, which is the name of the file. It would be nice to have a wrapper around the exception that adds the file name, but I'm not going to insist on a “pythonesque” approach in this PR.